### ISSUE TITLE
Eslint-plugin-vue goes crazy on normal js but not working in vue files

### ISSUE THREAD

**Issue by pingshunhuangalex** [vuejs/eslint-plugin-vue#80]
**Tell us about your environment**

* **ESLint Version:** eslint@4.1.1
* **eslint-plugin-vue Version:** eslint-plugin-vue@3.5.1
* **Node Version:** v6.10.2

**Please show your full configuration:**

```
// http://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parser: 'babel-eslint',
  parserOptions: {
    sourceType: 'module'
  },
  env: {
    browser: true,
  },
  // required to lint *.vue files
  plugins: [
    'html',
    'vue'
  ],
  extends: [
    'eslint:recommended',
    'plugin:vue/recommended'
  ],
  // add your custom rules here
  'rules': {
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
    'no-console': ['warn', { allow: ['warn', 'error'] }],
    'no-undef': 'off',
    'quotes': ['warn', 'single', {
      'avoidEscape': true,
      'allowTemplateLiterals': true
    }],
    'semi': ['warn', 'always']
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
A bunch of `eslint-plugin-vue` rules gets triggered in a normal js file, which is actually a mixin file I used for my global object for state management.

```vue (it's actually just a js)
import store from './store';

export default {
  data () {
    return {
      //...
    };
  },
  computed: {
    //...
  },
  methods: {
    //...
  }
};
```

**What did you expect to happen?**
`eslint-plugin-vue` is only triggered in `*.vue` files, or just wonder if there is a way to disable just the `eslint-plugin-vue` rules on normal .js

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint-plugin-vue goes crazy in Atom on normal js
![image](https://user-images.githubusercontent.com/17170199/28245327-574d1bfa-6a47-11e7-9001-873b6ac37b0a.png)

but not actually working in vue files (note I took the key out of the v-for, and I actually have v-for and v-if in the same component.)
![image](https://user-images.githubusercontent.com/17170199/28245350-da75e930-6a47-11e7-842c-739f4e3f3d1e.png)

**Comment by mysticatea** [comment]
Thank you for the report.

However, it seems an npm problem.
`eslint-plugin-vue` is using `vue-eslint-parser`, but in your environment, npm seems to install an old `vue-eslint-parser`. So the error message says "Use the latest vue-eslint-parser".

Please try to remove `node_modules` and `package-lock.json` then `npm install`. (and mind that re-start your editor.)

**Comment by pingshunhuangalex** [comment]
Thanks @mysticatea 
The errors do go away now, but it still doesn't quite work in vue files as shown in the screenshot above, as if it's not recognising my vue files. Is there anything else you can see from my settings? Thanks.

**Comment by mysticatea** [comment]
Is your editor configured that it runs eslint on `.vue` files?
I'm not familiar with Atom, but for example, VSCode does not run eslint on `.vue` files by default.

**Comment by pingshunhuangalex** [comment]
I did all the setup to get `babel-eslint` work in vue file, and I did notice that moving `parser: 'babel-eslint',` to `parserOptions` also breaks my normal babel js linting in vue file.

That was a good question, I actually not sure if Atom fully support running eslint on vue files yet. I guess I haven't fully understood the setup here, and I'm just gonna roll back to my previous setup for now. Hopefully there will be an official vue-linting package for vue in the future.

Thanks for the help.
